pandas/tests/indexes/test_base.py

```diff
@@ -499,25 +483,25 @@ def test_insert(self):
         null_index = Index([])
         tm.assert_index_equal(Index(['a']), null_index.insert(0, 'a'))
 
+    @pytest.mark.parametrize("na_val", [np.nan, pd.NaT, None])
+    def test_insert_missing(self, na_val):
         # GH 18295 (test missing)
         expected = Index(['a', np.nan, 'b', 'c'])
-        for na in (np.nan, pd.NaT, None):
-            result = Index(list('abc')).insert(1, na)
-            tm.assert_index_equal(result, expected)
-
-    def test_delete(self):
-        idx = Index(['a', 'b', 'c', 'd'], name='idx')
-
-        expected = Index(['b', 'c', 'd'], name='idx')
-        result = idx.delete(0)
+        result = Index(list('abc')).insert(1, na_val)
         tm.assert_index_equal(result, expected)
-        assert result.name == expected.name
 
-        expected = Index(['a', 'b', 'c'], name='idx')
-        result = idx.delete(-1)
-        tm.assert_index_equal(result, expected)
-        assert result.name == expected.name
+    @pytest.mark.parametrize("pos,exp", [
+        (0, Index(['b', 'c', 'd'], name='idx')),
+        (-1, Index(['a', 'b', 'c'], name='idx'))
+    ])
+    def test_delete(self, pos, exp):
```

Comment:
note that some of the tests here should really be parametreized on index types themselves, there is an open issue on that (eg. these set type ops, test_delete and such), are generally tested by the subclasses, but we need a more general cleanup on that